File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        ArrayList<Integer> row = new ArrayList<Integer>();
        ArrayList<Integer> col = new ArrayList<Integer>();
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if(matrix[i][j] == 0){
                    row.add(i);
                    col.add(j);
                }
            }
        }
        for(int i = 0; i<m;i++){
            for(int j = 0; j < n;j++){
                if(row.contains(i) || col.contains(j)){
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        col = len(matrix[0])
        row = len(matrix)
        rowZeroAt = []
        colZeroAt = []
        for i in range(row):
            for j in range(col):
                if matrix[i][j] == 0:
                    rowZeroAt.append(i)
                    colZeroAt.append(j)

        for i in range(row):
            for j in range(col):
                if i in rowZeroAt or j in colZeroAt:
                    matrix[i][j] = 0
        
        return matrix
